lib/src/data/group_class.dart

```diff
@@ -0,0 +1,47 @@
+import "utils.dart";
+
+enum GroupStatus {
+  openToAll,
+  openByRequest,
+  private,
+}
+
+class Group {
+  final String id;
+  final String name;
+  final List<String> members; 
+  int? memberLimit;
+  GroupStatus status;
```

Comment:
```suggestion
extension type GroupID(String value) { }

class Group {
  final GroupID id;
  final List<UserID> members; 

  String name;
  int? memberLimit;
  GroupStatus status;
```

- using GroupIDs and UserIDs
- making `name` non-final and separating final vs non-final fields